### PR TITLE
Allstops retry button

### PR DESCRIPTION
**Pull request by ywsang**
#144

### CODE REVIEW

File: TCAT/Controllers/AllStopsTableViewController.swift [hunk 1]
```diff
@@ -185,4 +192,36 @@ class AllStopsTableViewController: UITableViewController, DZNEmptyDataSetSource,
...
+    
+            self.tableView.reloadData()
+        }
+    }
+    
+    func retryNetwork(completion: @escaping () -> Void) {
+        Network.getAllStops().perform(withSuccess: { stops in
+            let allBusStops = stops.allStops
+            if !allBusStops.isEmpty {
+                //Only updating user defaults if retriving from network is successful
```
Comment (myo3) [inline TCAT/Controllers/AllStopsTableViewController.swift]: space after `//`

File: TCAT/Controllers/AllStopsTableViewController.swift [hunk 2]
```diff
@@ -185,4 +192,36 @@ class AllStopsTableViewController: UITableViewController, DZNEmptyDataSetSource,
...
+                let data = NSKeyedArchiver.archivedData(withRootObject: allBusStops)
+                userDefaults.set(data, forKey: Constants.UserDefaults.allBusStops)
+            }
+            completion()
+        }, failure: { error in
+            print("retryNetwork error:", error)
```
Comment (myo3) [inline TCAT/Controllers/AllStopsTableViewController.swift]: Write class name too so know which file print statement is coming from

File: TCAT/Controllers/AllStopsTableViewController.swift [hunk 3]
```diff
@@ -185,4 +192,36 @@ class AllStopsTableViewController: UITableViewController, DZNEmptyDataSetSource,
...
     }
+    
+    func buttonTitle(forEmptyDataSet scrollView: UIScrollView!, for state: UIControlState) -> NSAttributedString! {
+        let title = "Retry"
+        let attrs = [NSAttributedStringKey.foregroundColor: UIColor.buttonColor]
+        return NSAttributedString(string: title, attributes: attrs)
+    }
+    
+    func emptyDataSet(_ scrollView: UIScrollView!, didTap button: UIButton!) {
+        retryNetwork{() -> Void in
```
Comment (myo3) [inline TCAT/Controllers/AllStopsTableViewController.swift]: make the code in the completion block a separate function so everything in the `emptyDataSet` function reads at the same level of abstraction

File: TCAT/Utilities/SearchTableViewHelpers.swift [hunk 4]
```diff
@@ -59,15 +59,18 @@ class SearchTableViewManager {
 
     private init() {}
     func getAllStops() -> [BusStop] {
-        if let stops = allStops {
-            return stops
+        if let stops = allStops{
+            // Check if not empty so that an empty array isn't returned 
+            if stops != [] {
```
Comment (myo3) [inline TCAT/Utilities/SearchTableViewHelpers.swift]: use `isEmpty`

File: TCAT/Controllers/AllStopsTableViewController.swift [hunk 5]
```diff
@@ -109,7 +104,19 @@ class AllStopsTableViewController: UITableViewController, DZNEmptyDataSetSource,
         }
         
         return sectionIndexDictionary
-
+    }
+    
+    /// Retrieves the keys from the sectionIndexDictionary
```
Comment (myo3) [inline TCAT/Controllers/AllStopsTableViewController.swift:105]: put this function in the table data source section so every function is under a `MARK`-ed section

**Review by myo3 — APPROVED** [review]
Looks good! Added some stylistic comments

**Review by mattbarker016 — COMMENTED** [review]
Looks good to me, just address everything Monica mentioned!

File: TCAT/Controllers/AllStopsTableViewController.swift (hunk 5 above)
Comment (ywsang) [reply TCAT/Controllers/AllStopsTableViewController.swift:105]: Should I move the other similar function into the table data source section as well?

Comment (myo3) [reply TCAT/Controllers/AllStopsTableViewController.swift:105]: if there's a lot maybe make a separate section for these data functions. if it's not that many then just put them in table data source. goal is just to organize functions by section

**Review by mattbarker016 — APPROVED** [review]

**Pull request merged**